### ISSUE TITLE
[Elixir] Implement new Concept Exercise: multiple-clause-functions

### ISSUE THREAD

**Issue by neenjaw** [exercism/v3#1365]
This issue describes how to implement the `functions` concept exercise for the elixir track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal for this exercise is to teach more about functions in elixir and how to use multiple functions clauses, guards

## Learning objectives

A student upon completion will:

- Know what a guard is.
- Know how to use a guard.
- Know that guards return strict boolean values.
- Know that guards meet certain guarantees to not mutate state.
- Know that guards must be declared in a special form, and not all functions that return boolean values are guards.
- Know how to write basic multiple-clause functions.
- Know how to use guards to protect execution of a function.
- Know how to use guards to perform basic control flow.
- Know how to specify a default arguemnt
- Know that by specifying a default argument, it defines multiple arities for the function name

## Out of scope

- pattern matching and multiple-clause functions

## Concepts

- `multiple-clause-functions`
- `guards`
- `default-parameters`

## Prerequisites

- `basics`
- `booleans`
- `conditionals`
- ? `anonymous-functions`

## Resources to refer to

- [multiple clauses in anon functions](https://til.hashrocket.com/posts/36c6d2684e-defining-multiple-clauses-in-an-anonymous-function)
- https://elixirschool.com/en/lessons/basics/functions/#guards
- https://elixirschool.com/en/lessons/basics/functions/#default-arguments

## Representer

No modifications at this time

## Analyzer

No analyzer features at this time.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by neenjaw** [comment]
@fireproofsocks as we discussed, maybe we could get this going at somepoint before pattern matching to make it a bit better the pattern-matching exercise

**Comment by fireproofsocks** [comment]
Yes @neenjaw  -- I think this one will dovetail with the tail-recursion stuff because tail-recursive functions will always be calling variants of themselves.
